### DISCUSSION TITLE
Building Dockerfile with CUDA support and smaller image.

### DISCUSSION THREAD

**Discussion by jflachman** [abetlen/llama-cpp-python#1609] [Q&A] [answered]
I had two issues trying to build a file with CUDA support.  

1. I could not build with llama-cpp-pytyon:2.82.  I had to revert to 2.77.  I also reverted to CUDA 12.1.1.
2. The docker image size is a little more than 13GB.  I wanted a smaller docker image.

After much try and error, I solved 1 by reverting to build 2.77.  I used the approach for @Hansson0728 had for issue #1119. https://github.com/abetlen/llama-cpp-python/issues/1119.  However, that still did not work completely for me.  It turns out I needed to set `LD_LIBRARY_PATH` for the CUDA libraries.  The path in the `devel` image had also changed since he created that solution.

I have attached the working Dockerfile below.

**Comment by jflachman** [comment, marked as answer]
**Solved**

I finally solved the problem.  I rolled back to the version of `llama-cpp-python/docker/cuda-simple/Dockerfile` from before update `#1530 `made 3 weeks ago.  This used a base image of `nvidia/cuda/12.1.1-devel-ubuntu22.04` and I used `llama-cpp-python==0.2.77` with the associate whl file.

**Reply by jflachman** [reply]
After getting a working CUDA build (13GB), I updated the Dockerfile for multistep build.  This reduced the build to 5GB.

Here is the Dockerfile that works. (5GB image)

[Dockerfile.txt](https://github.com/user-attachments/files/16327463/Dockerfile.txt)

You may also download the docker container here:  [Docker Hub:  jflachman/llama-cpp-python](https://hub.docker.com/r/jflachman/llama-cpp-python).  This also includes a brief guide for its use.  Using this container will allow you to provide a `server.config` file on startup to support _server configuration_ and _multiple model setup_.

**Reply by p1x31** [reply]
```
COPY --from=builder root/.local/lib/python3.10/site-packages/llama_cpp /usr/local/lib/python3.10/dist-packages/llama_cpp
COPY --from=builder root/.local/lib/python3.10/site-packages/lib /usr/local/lib/python3.10/dist-packages/lib
COPY --from=builder root/.local/lib/python3.10/site-packages/include /usr/local/lib/python3.10/dist-packages/include
```

These lines are broken
